install/cupy_builder/cupy_setup_build.py

```diff
@@ -343,15 +343,23 @@ def make_extensions(ctx: Context, compiler, use_cython):
             # that are executable in the build environment, not the target
             # environment.
             compiler = ccompiler.new_compiler()
+            sysconfig.customize_compiler(compiler)
+            # Need to match and replace these
+            # https://github.com/pypa/distutils/blob/30b7331b07fbc404959cb37ac311afdfb90813be/distutils/unixccompiler.py#L117-L129
             cc = os.environ['CC_FOR_BUILD' if PLATFORM_LINUX else 'CC']
             cxx = os.environ['CXX_FOR_BUILD' if PLATFORM_LINUX else 'CXX']
             compiler.compiler = [cc,]
-            compiler.compiler_cxx = [cxx,]
             compiler.compiler_so = [cc,]
-            compiler.linker_exe = [cc, f'-B{os.environ["BUILD_PREFIX"]}/bin']
+            compiler.compiler_cxx = [cxx,]
+            compiler.compiler_so_cxx = [cxx,]
             compiler.linker_so = [cc, f'-B{os.environ["BUILD_PREFIX"]}/bin',
                                   '-shared']
-
+            compiler.linker_so_cxx = [cxx, f'-B{os.environ["BUILD_PREFIX"]}/bin',
+                                      '-shared']
+            compiler.linker_exe = [cc, f'-B{os.environ["BUILD_PREFIX"]}/bin']
+            compiler.linker_exe_cxx = [cxx, f'-B{os.environ["BUILD_PREFIX"]}/bin']
```

Comment:
These were resorted to match [the order in Distutils]( https://github.com/pypa/distutils/blob/30b7331b07fbc404959cb37ac311afdfb90813be/distutils/unixccompiler.py#L117-L129 ), which should hopefully make it easier to compare and update when needed